components/changelog/ChangelogTrigger.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { gql } from '@apollo/client';
+import { graphql, withApollo } from '@apollo/client/react/hoc';
+import themeGet from '@styled-system/theme-get';
+import { cloneDeep } from 'lodash';
+import styled from 'styled-components';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import { parseToBoolean } from '../../lib/utils';
+
+import { Flex } from '../Grid';
+import { withNewsAndUpdates } from '../NewsAndUpdatesProvider';
+import { Dropdown } from '../StyledDropdown';
+import { useUser } from '../UserProvider';
+
+import ChangelogNotificationDropdown from './ChangelogNotificationDropdown';
+
+const CHANGE_LOG_UPDATES_ENABLED = parseToBoolean(process.env.CHANGE_LOG_UPDATES_ENABLED);
+
+const FlameIcon = styled(Flex)`
+  border-radius: 50%;
+  background-color: ${props => themeGet(props.backgroundColor)};
+  background-image: ${props => `url(${props.url})`};
+  background-repeat: no-repeat;
+  background-position: center center;
+  height: 30px;
+  width: 30px;
+  margin-left: 2px;
+`;
+
+const ChangelogTrigger = props => {
+  const { setShowNewsAndUpdates, setChangelogViewDate } = props;
+  const { LoggedInUser } = useUser();
+  const LoggedInUserFromCache = props.client.readQuery({ query: loggedInUserQuery })?.LoggedInUser || LoggedInUser;
+  const hasSeenNewUpdates = LoggedInUserFromCache?.hasSeenLatestChangelogEntry;
+
+  const handleShowNewUpdates = () => {
+    setShowNewsAndUpdates(true);
+    setChangelogViewDate({ variables: { changelogViewDate: new Date() } });
```

Comment:
The cache update is usually done in the `update` function of the mutation (we only want to do it if it succeeds).

See: https://www.apollographql.com/docs/react/data/mutations/#options. And there's an example in https://github.com/opencollective/opencollective-frontend/blob/5710519f07b30f4c2872eed4c3b737ac891fc6fc/components/collective-page/sections/Contribute.js#L132

Reply:
@Betree : Fixed this. Let me know if all looks correct here. 😄